docs/tutorial/visualization.rst

```diff
@@ -0,0 +1,420 @@
+
+.. code-block:: python
+
+    from compas.artists import Artist
+    from compas.geometry import Frame, Box
+
+    box = Box(Frame.worldXY(), 1, 1, 1)
+
+    artist = Artist(box)
+    artist.draw()
+
+
+When running this code snippet in Rhino (using ``RunPythonScript``) the created artist will be of type :class:`compas_rhino.artists.BoxArtist`.
+
+.. code-block:: python
+
+    print(type(artist))
+    # <class 'compas_rhino.artists.boxartist.BoxArtist'>
+
+
+The base artist detected that the current execution context is Rhino,
+and that the object passed to the constructor is a :py:class:`compas.geometry.Box`.
+
+In Blender, we will get a :class:`compas_blender.artists.BoxArtist`.
+
+.. code-block:: python
+
+    print(type(artist))
+    # <class 'compas_blender.artists.boxartist.BoxArtist'>
+
+
+In Grasshopper, the snippet can be executed from within a GH User Component, with a small modification.
+The output of the draw function has to be assigned to the output variable of the component (e.g. output variable ``a``).
+
+.. code-block:: python
+
+    from compas.artists import Artist
+    from compas.geometry import Frame, Box
+ 
+    box = Box(Frame.worldXY(), 1, 1, 1)
+
+    artist = Artist(box)
+    a = artist.draw()
+
+
+.. code-block:: python
+
+    print(type(artist))
+    # <class 'compas_ghpython.artists.boxartist.BoxArtist'>
+
+
+Colors
+======
+
+Colors are in the form of RGB tuples.
+In Rhino and Grasshopper the components have to be specified in integer format with values between 0 and 255.
+In Blender, colors have RGB components in float format with values between 0 and 1.
+Values beyond 1 will cause the object to emit light with an intensity related to the provided number.
+Therefore, the same colors can be used in Rhino, Grasshopper, and Blender, if they are in 0-255 integer format.
+
+.. code-block:: python
+
+    from compas.geometry import Frame, Box
+    from compas.artists import Artist
+
+    box = Box(Frame.worldXY(), 1, 1, 1)
+
+    Artist(box).draw(color=(0, 255, 0))
+
+
+Primitives and shapes only have one color attribute: ``compas.artists.PrimitiveArtist.color`` and ``compas.artists.ShapeArtist.color``.
+This attribute can be set when the artist is constructed by providing a value for the parameter ``color``,
+or by assigning a value to the attribute afterwards.
+
+.. code-block:: python
+
+    artist = Artist(box, color=(255, 0, 0))
+
+    artist.color = (0, 0, 255)
+
+    artist.draw()
+
+
+The color value stored in the ``color`` attribute can be temporarily overwritten
+using the ``color`` parameter of the ``draw`` function.
+Note that the value of this parameter is not stored in the ``color`` attribute
+and therefore only has an effect on the specific ``draw`` call.
+
+.. code-block:: python
+
+    artist.draw(color=(0, 255, 0))
+
+
+Data structure artists a few mor color settings in addition to ``color``.
```

Comment:
Something's missing here... ;) Did you mean `Data structure artists have a few more [...]` ?

Reply:
haha, no, not really, but i guess the statement is not very informative....